tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/target/TargetPlatformFactoryImpl.java

```diff
@@ -186,18 +182,26 @@ public P2TargetPlatform createTargetPlatform(TargetPlatformConfigurationStub tpC
         applyConfiguredFilter(filter, reactorProjectUIs.keySet());
         applyFilters(filter, externalUIs, reactorProjectUIs.keySet(), eeResolutionHandler.getResolutionHints());
 
+        IRawArtifactFileProvider externalArtifactFileProvider = createExternalArtifactProvider(completeRepositories,
+                targetFileContent, pomDependenciesContent.getArtifactRepoOfPublishedBundles(),
+                tpConfiguration.getIncludePackedArtifacts());
         PreliminaryTargetPlatformImpl targetPlatform = new PreliminaryTargetPlatformImpl(reactorProjectUIs, //
                 externalUIs, //
                 pomDependenciesContent.getMavenInstallableUnits(), //
                 eeResolutionHandler.getResolutionHints(), //
                 filter, //
                 localMetadataRepository, //
-                createExternalArtifactProvider(completeRepositories, targetFileContent, pomDependencyArtifactRepo,
-                        tpConfiguration.getIncludePackedArtifacts()), //
+                externalArtifactFileProvider, //
                 localArtifactRepository, //
                 includeLocalMavenRepo, //
                 logger);
 
+        RepositoryBlackboardKey blackboardKey = RepositoryBlackboardKey
```

Comment:
Can we add a little description what this blackboardkey is for and what it does (beside that some test fails if it is not there...). Is it some legacy code we might remove in the future?